### ISSUE TITLE
Replace all don't replace ALL

### ISSUE THREAD

**Issue by tbayart** [notepad-plus-plus/notepad-plus-plus#4768]
### Description of the Issue
When using "Replace" functionality, clicking "Replace All" button replace all occurences from the cursor.
If the cursor is in the middle of the text, it does not replace all.

### Steps to Reproduce the Issue
1. Input "test" text
2. Duplicate the line 10 times to get 10 identical lines
3. Move your cursor to line 5
4. Open "Replace" panel
5. Put "test" in "Find what", "zz" in "Replace with"
6. Click "Replace All"

### Expected Behavior
All lines should be replaced with "zz"

### Actual Behavior
Only lines from cursor to the end are repalced with "zz", previous lines kept "test"

### Debug Information
Notepad++ v7.5.8   (32-bit)
Build time : Jul 23 2018 - 02:03:53
Path : C:\Program Files (x86)\Notepad++\notepad++.exe
Admin mode : ON
Local Conf mode : OFF
OS : Windows 7 (64-bit)
Plugins : docMonitor.unicode.dll HexEditor.dll HTMLTag_unicode.dll NppFTP.dll NPPJSONViewer-win32.dll PluginManager.dll XMLTools.dll DSpellCheck.dll mimeTools.dll NppConverter.dll NppExport.dll 

![image](https://user-images.githubusercontent.com/5262768/44197406-c3cf7f00-a13e-11e8-8abd-c65cf7c8575f.png)

**Comment by ClaudiaFrank** [comment]
Actually I think it should do exactly what it does, to get what you want you just need to check the 
Wrap around checkbox.

**Comment by tbayart** [comment]
Thanks for the "Wrap around" trick.
Without this checked, it does not replace all as it should.

**Comment by ClaudiaFrank** [comment]
>Without this checked, it does not replace all as it should.

I think it does what it has to, taking into accout that there is this checkbox available.
Also the find function works this way - from cursor position to either start or end of document,
if whole document should be affected, check wrap around.
I disagree saying this is a bug and think this issue can be closed to be honest.

**Comment by tbayart** [comment]
The find function don'twork this way.
No matter the "Wrap around" is checked or not, if you click "Find all in current document" it will find all occurence, before AND after the cursor.

**Comment by tbayart** [comment]
Same behavior for the "Count" button in Find function.
No matter the "Wrap around" is checked, the result is the same.

**Comment by ClaudiaFrank** [comment]
Sorry, I didn't make clear that I was referring to the find next (or the << and >>) button(s) within the replace tab.

**Comment by dinkumoil** [comment]
Though you can't imagine, there **are** people around who really **need** the functionality which you call a bug. Please believe it and close this issue.

**Comment by tbayart** [comment]
So you mean you want to replace all but not all...
You can use the "In selection" checkbox to limit the replace all functionality.
It's more logical.

**Comment by dinkumoil** [comment]
The "Wrap around" checkbox is checked in a fresh Notepad++ installation, you already got your "logical" behaviour of the "Search & Replace" dialog.

This checkbox is intended for some special situations where it is necessary to do a replace operation from the current cursor position to e.g. EOF only. If you toggle the state of this checkbox and get unexpected results it is a RTFM problem and not worth to report a bug. Please close this issue.

**Comment by DonBarredora** [comment]
Maybe this difference in behaviour should be documented somewhere.

**Comment by dinkumoil** [comment]
@DonBarredora 
Well, it **is** documented **somewhere**...

At the [homepage of Notepad++](https://notepad-plus-plus.org/) under "resources" (left menu bar) you can find a link to the [Notepad++ wiki](http://docs.notepad-plus-plus.org/index.php/Main_Page).

There, the link under (left menu bar) "NP++ ARTICLES -> byTopic -> F -> Find and replace" finally leads you to the page [How to find, replace and highlight text in Notepad++](http://docs.notepad-plus-plus.org/index.php/Searching_And_Replacing).

Also on the wiki's main page, under (left menu bar) "NP++ RESOURCES -> Guides", you can find a [link to a page](http://docs.notepad-plus-plus.org/index.php/NP%2B%2B_Guides) where you can download the user manual for Notepad++ v6.6.4 as CHM file (click on "User manual - v6.6.4 (CHM)").

After downloading the file maybe (it depends on your web browser) you have to unblock it because Windows Explorer considers it to be an unsafe file from the internet. To do that go to your download folder, right-click the downloaded file and click on "Unblock" and then "Apply".

![grafik](https://user-images.githubusercontent.com/24699812/44306798-36866780-a396-11e8-9f28-9dd7b310dbee.png)

Open the CHM file by double-clicking it, navigate to chapter "Searching" and open subchapter "Normal Search".

I know, you will say "Naah, it's an outdated file, hidden in an outdated wiki full of broken links!" and you are right.

The broken links are there because in the beginning the Notepad++ project was hosted on  sourceforge.net for a long time. From mid-2013 SourceForge introduced a program called DevShare, which offered projects a way to monetize their downloads by having an optional download that includes prompts for the user to download additional software that is not part of the project (the program was cancelled in February 2016 after SourceForge's owner had changed). 

As a consequence of this the Notepad++ team decided to leave SourceForge. Unfortunately the wiki respectively the contained links have never been ported completely to the new location, a lot of links still lead to SourceForge pages which are not existing anymore.

Notepad++ is open source and is given to you for free, don't look a gift horse in the mouth. Also feel free to improve the situation and update the wiki respectively the help file mentioned above.

**Comment by HormigaInformatica** [comment]
I come here thinking that is a bug too... the actual behaviour was to replace(find/mark/count) in whole file... I think it could be better the check "Replace All in Direction" instead a check with "wrap around" ... I have already installed notepad++ and the checkbox "wrap around" was not checked... I wasted some time trying to guess what is happening, updating all, reading changelog, also I tried to open an issue (thinking in a bug) because the usual behaviour changed.

Thanks for the info about how to search.

**Comment by sasumner** [comment]
>I come here thinking that is a bug too

I guess it comes down to you have to learn the programs you use, and their features.  I can't remember exactly the verbage used on the UI for the editor I used before Notepad++, but when I saw "Wrap around" I knew immediately what that meant.  So I don't understand why people seem to have a problem with this.

**Comment by tbayart** [comment]
> I guess it comes down to you have to learn the programs you use

very clever
> I can't remember exactly the verbage used on the UI for the editor

There is previous messages in this thread, please read from the beginning
> So I don't understand why people seem to have a problem with this.

As previously explained, "Replace All" should means replace all but it doesn't, it depend on "Wrap around", so it mean "Replace to the end" not replace all
Also some other functions doesn't care of the "Wrap around" option (ie Find all in current document)

**Comment by sasumner** [comment]
>very clever

Thanks, I guess.  The original statement was totally true, though, so I'm not sure what this is supposed to mean.

>I can't remember exactly the verbage used on the UI for the editor

I'm talking about the editor I used for many years before Notepad++, not Notepad++ itself (so no need to read from the beginning...which I have done since this issue was created).  Actually, if you'd have read (and mentally processed) my whole sentence, it started out "I can't remember exactly the verbage used on the UI for the editor _I used before Notepad++_..."

>Replace to the end

As previously explained in this thread (maybe you should read it from the beginning to the end), there isn't enough room on the button for a whole lot of text, especially when non-English translations are considered as a whole.

**Comment by tbayart** [comment]
> I'm talking about the editor I used for many years before Notepad++, not Notepad++ itself

Got it, sorry for the misunderstanding
